File: control-panel/website-builder/preview/api.call.py
import os
import subprocess

# Change to the desired directory where your PHP files are located
os.chdir(r"C:\xampp\htdocs\online-store.varsitymarket.package\website-builder\preview\docs")

# Command to start the PHP built-in server
command = [
    r"C:\xampp\php\php.exe",  # Replace with the path to your PHP executable
    "-S",
    "127.0.0.1:8000"  # Change the port if needed
]

# Open a log file to capture the output
with open("session.log.hash.report.pxy", "w") as log_file:
    subprocess.run(command, stdout=log_file, stderr=subprocess.STDOUT)

File: control-panel/website-builder/preview/api.call.php
<?php
// Set the directory you want to serve
$directory = 'C:\xampp\htdocs\online-store.varsitymarket.package\website-builder\preview\docs';
#$directory = 'C:\xampp\htdocs\online-store.varsitymarket.package\trash\template-sameple';


// Change to the specified directory
chdir($directory);

// Define the command to start the PHP built-in server
$command = 'php -S 192.168.72.41:700';

// Open a log file to capture the output
$logFile = fopen("session.log.hash.report.pxy", "w");

// Start the server process and redirect output to the log file
$process = popen($command, 'r');

if (is_resource($process)) {
    // Log the output to the log file
    while (!feof($process)) {
        $output = fgets($process);
        fwrite($logFile, $output);
        echo $output; // Optionally output to console as well
    }
    pclose($process);
}

fclose($logFile);
?>